Miscellaneous/practice.py

```diff
@@ -0,0 +1,32 @@
+def vowels(sentence):
+    letter_a = 0
+    letter_e = 0
+    letter_i = 0
+    letter_o = 0
+    letter_u = 0
+
+    #sentence = 'The boy is coming'
+    for word in sentence:
+        if  word == "a":
+            letter_a += 1
+
+        if word == 'e':
+            letter_e += 1
+        
+        if word == 'i':
+            letter_i += 1
+
+        if word == 'o':
+            letter_o += 1
+
+        if word == 'u':
+            letter_u += 1   
+
+    result = { "Count of a" : letter_a, "Count of e" : letter_e, "Count of i" : letter_i, "Count of o" : letter_o, "Count of u" : letter_u }
```

Comment:
So here are a few suggestions to improve this logic and also to drastically shorten the number of lines involved. 

1. Rather assigning each vowel letter as a separate variable, you can use a dictionary like you do here on line 25, such that in your loop, you'll add to the values of the respective keys in the dictionary, like so:

```python
def vowels(sentence):
    vowels_dict = { "a": 0, "e": 0, "i": 0, "o": 0, "u": 0}

    for letter in sentence:
        if letter  == "a":
            vowels_dict["a"] += 1
        elif letter == "e":
            vowels_dict["e"] += 1
        elif letter == "i":
            vowels_dict["e"] += 1
        etc...
    return vowels_dict
```
Also, see how I'm using `if-elif` blocks instead of just separate `if` blocks? We don't need to check the same letter for all the conditionals if we've already determined whether or not it is a vowel and the correct sum to add it to.

2. Given how the `vowels_dict`dictionary is setup (just using the actual vowels as the keys), you can take things a step further and condense all those conditional checks for if a letter == a vowel, into just 1 conditional, like so:

```python
def vowels(sentence):
    vowels_dict = { "a": 0, "e": 0, "i": 0, "o": 0, "u": 0}

    for letter in sentence:
       if letter in vowel_dict:
            vowels_dict[letter] += 1
    return vowels_dict
```

3. Lastly, to ensure that this function counts both upper case and lower case vowels, we can convert the input string into a lower case string using the lower() method:

```python
def vowels(sentence):
    vowels_dict = { "a": 0, "e": 0, "i": 0, "o": 0, "u": 0}

    for letter in sentence.lower():
        if letter in vowel_dict:
            vowels_dict[letter] += 1
    return vowels_dict
```